### PR TITLE
Refactor errors in apps

### PR DESCRIPTION
**Pull request by duanxianbo**
(task:13688)

### CODE REVIEW

File: apps/dao/sqldb/github_code_review.go [hunk 1]
```diff
@@ -45,24 +45,14 @@ func (g GithubCodeReview) FindCodeReviewByGithubReviewerID(
 			&codeReview.Round,
 		)
 
-	if errors.Is(err, sql.ErrNoRows) {
-		internalErr := &errs.Error{
-			Code: errs.NotFound,
-			Message: fmt.Sprintf(
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return entity.GithubCodeReview{}, errs.NewError(errs.NotFound, fmt.Sprintf(
```
Comment (naturetime10) [inline apps/dao/sqldb/github_code_review.go]:
```suggestion
			return entity.GithubCodeReview{}, errs.NewError(
			    errs.NotFound, 
			    fmt.Sprintf(
```

File: apps/dao/sqldb/github_code_review.go [hunk 2]
```diff
@@ -91,24 +81,15 @@ func (g GithubCodeReview) FindCodeReviewByInternalTaskID(
...
-		g.dataCollector.Logger.ErrorWithContext(ct, internalErr)
-		return entity.GithubCodeReview{}, internalErr
-	}
 
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:     errs.Unknown,
-			EmbedErr: err,
+		if errors.Is(err, sql.ErrNoRows) {
+			return entity.GithubCodeReview{}, errs.NewError(errs.NotFound, fmt.Sprintf(
```
Comment (naturetime10) [inline apps/dao/sqldb/github_code_review.go]:
```suggestion
			return entity.GithubCodeReview{}, errs.NewError(
			    errs.NotFound, 
			    fmt.Sprintf(
```

File: apps/github/app_api.go [hunk 3]
```diff
@@ -108,25 +108,15 @@ func (a AppAPI) webInstall(writer http.ResponseWriter, request *http.Request) {
...
 	teamID, err := strconv.ParseUint(teamIDRaw, 10, 64)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:     errs.InvalidArgument,
-			EmbedErr: err,
-			Message:  "must provide teamId",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "must provide teamId")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:113]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 4]
```diff
@@ -108,25 +108,15 @@ func (a AppAPI) webInstall(writer http.ResponseWriter, request *http.Request) {
...
 	redirectURL := query.Get("redirectUrl")
 	if len(redirectURL) == 0 {
-		internalErr := &errs.Error{
-			Code:     errs.InvalidArgument,
-			EmbedErr: err,
-			Message:  "must provide redirectUrl",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "must provide redirectUrl")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:122]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 5]
```diff
@@ -148,14 +137,12 @@ func (a AppAPI) webInstall(writer http.ResponseWriter, request *http.Request) {
 	}
 	internalErr := a.githubAppInstallStateDao.CreateState(ct, state)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 6]
```diff
@@ -148,14 +137,12 @@ func (a AppAPI) webInstall(writer http.ResponseWriter, request *http.Request) {
...
 		return
 	}
 
 	installURL, internalErr := a.getInstallGithubAppURL(ct, state.ID)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 7]
```diff
@@ -169,56 +156,36 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
...
 	stateID, err := strconv.ParseUint(stateIDParam, 10, 64)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:     errs.InvalidArgument,
-			EmbedErr: err,
-			Message:  "fail to parse state ID",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "fail to parse state ID")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:166]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 8]
```diff
@@ -169,56 +156,36 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
...
 	rawInstallationID := query.Get("installation_id")
 	if len(rawInstallationID) == 0 {
-		internalErr := &errs.Error{
-			Code:     errs.InvalidArgument,
-			EmbedErr: err,
-			Message:  "must provide installation_id",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "must provide installation_id")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:174]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 9]
```diff
@@ -169,56 +156,36 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
...
 	installationID64, err := strconv.ParseInt(rawInstallationID, 10, 32)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:     errs.InvalidArgument,
-			EmbedErr: err,
-			Message:  "installation_id must be int",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "installation_id must be int")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:182]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 10]
```diff
@@ -169,56 +156,36 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
...
 	expireAt := state.CreatedAt.Add(a.config.InstallationValidDuration)
 	now := time.Now().UTC()
 	if expireAt.Before(now) {
-		internalErr = &errs.Error{
-			Code:    errs.InvalidOperation,
-			Message: "install app session expired",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidOperation, "install app session expired")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:199]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 11]
```diff
@@ -169,56 +156,36 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
...
+		internalErr := errs.NewError(errs.InvalidArgument, "installation_id must be int")
 		errs.SetHTTPErr(internalErr, writer)
 		return
 	}
 
 	installationID := int(installationID64)
 	state, internalErr := a.githubAppInstallStateDao.FindStateByID(ct, stateID)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 12]
```diff
@@ -230,14 +197,12 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
 	}
 	internalErr = a.githubAppInstallationDao.CreateGithubAppInstallation(ct, ins)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 13]
```diff
@@ -230,14 +197,12 @@ func (a AppAPI) webFinishInstall(writer http.ResponseWriter, request *http.Reque
...
 		return
 	}
 
 	internalErr = a.githubAppInstallStateDao.DeleteState(ct, stateID)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 14]
```diff
@@ -253,54 +218,33 @@ func (a AppAPI) webOnEventNotify(writer http.ResponseWriter, request *http.Reque
 	bodySignatureHeader := request.Header.Get("X-Hub-Signature-256")
 	bodySignatureHeaderParts := strings.Split(bodySignatureHeader, "=")
 	if len(bodySignatureHeaderParts) != 2 {
-		internalErr := &errs.Error{
-			Code:    errs.InvalidArgument,
-			Message: "signature header must have 2 parts",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "signature header must have 2 parts")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:235]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 15]
```diff
@@ -253,54 +218,33 @@ func (a AppAPI) webOnEventNotify(writer http.ResponseWriter, request *http.Reque
...
 	}
 
 	if bodySignatureHeaderParts[0] != "sha256" {
-		internalErr := &errs.Error{
-			Code:    errs.InvalidArgument,
-			Message: "signature header must start with sha256",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "signature header must start with sha256")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:242]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 16]
```diff
@@ -253,54 +218,33 @@ func (a AppAPI) webOnEventNotify(writer http.ResponseWriter, request *http.Reque
...
 	buf, err := io.ReadAll(request.Body)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:     errs.IO,
-			EmbedErr: err,
-			Message:  "fail to read request payload",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.IO, "fail to read request payload")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:250]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 17]
```diff
@@ -253,54 +218,33 @@ func (a AppAPI) webOnEventNotify(writer http.ResponseWriter, request *http.Reque
...
 
 	signature, err := hex.DecodeString(bodySignatureHeaderParts[1])
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:    errs.Unknown,
-			Message: "fail to decode request body signature",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "fail to decode request body signature")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:258]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 18]
```diff
@@ -253,54 +218,33 @@ func (a AppAPI) webOnEventNotify(writer http.ResponseWriter, request *http.Reque
...
 	}
 
 	if !validateHMACSignature(buf, []byte(a.config.WebhookSecret), signature) {
-		internalErr := &errs.Error{
-			Code:    errs.InvalidArgument,
-			Message: fmt.Sprintf("invalid request body signature: signature=%v", bodySignatureHeaderParts[1]),
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, fmt.Sprintf("invalid request body signature: signature=%v", bodySignatureHeaderParts[1]))
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:265]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 19]
```diff
@@ -321,39 +264,29 @@ func (a AppAPI) webListRequiredActionsForCurrentUser(writer http.ResponseWriter,
 	ct := request.Context()
 	userID, ok := ctx.UserIDFromContext(ct)
 	if !ok {
-		internalErr := &errs.Error{
-			Code:    errs.Unauthenticated,
-			Message: "user id not found",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.Unauthenticated, "user id not found")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:287]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 20]
```diff
@@ -309,7 +253,6 @@ func (a AppAPI) webOnEventNotify(writer http.ResponseWriter, request *http.Reque
 	a.dataCollector.Logger.InfoWithContext(ct, fmt.Sprintf("received event: deliveryID=%v EventType=%v", deliveryID, evtType))
 	internalErr := a.processEvent(ct, githubEntity.EventType(evtType), buf)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 21]
```diff
@@ -321,39 +264,29 @@ func (a AppAPI) webListRequiredActionsForCurrentUser(writer http.ResponseWriter,
...
 	teamIDRaw := chi.URLParam(request, teamIDParam)
 	teamID, err := strconv.ParseUint(teamIDRaw, 10, 64)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:    errs.InvalidArgument,
-			Message: "must provide teamId",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "must provide teamId")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:296]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 22]
```diff
@@ -321,39 +264,29 @@ func (a AppAPI) webListRequiredActionsForCurrentUser(writer http.ResponseWriter,
...
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
 		return
 	}
 
 	// TODO: receive notification from cloud and update required action status
 	requiredUserActions, internalErr = a.refreshRequiredActionsStatus(ct, userID, requiredUserActions)
 	if err != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 23]
```diff
@@ -368,23 +301,15 @@ func (a AppAPI) webCreateRequiredAction(writer http.ResponseWriter, request *htt
 	ct := request.Context()
 	requestSenderID, ok := ctx.UserIDFromContext(ct)
 	if !ok {
-		internalErr := &errs.Error{
-			Code:    errs.Unauthenticated,
-			Message: "user id not found",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.Unauthenticated, "user id not found")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:328]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 24]
```diff
@@ -368,23 +301,15 @@ func (a AppAPI) webCreateRequiredAction(writer http.ResponseWriter, request *htt
...
 	teamIDRaw := chi.URLParam(request, teamIDParam)
 	teamID, err := strconv.ParseUint(teamIDRaw, 10, 64)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:    errs.InvalidArgument,
-			Message: "must provide teamId",
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.InvalidArgument, "must provide teamId")
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:337]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 25]
```diff
@@ -395,20 +320,14 @@ func (a AppAPI) webCreateRequiredAction(writer http.ResponseWriter, request *htt
 	}{}
 	buf, err := io.ReadAll(request.Body)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code: errs.IO,
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.IO, err.Error())
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:349]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 26]
```diff
@@ -395,20 +320,14 @@ func (a AppAPI) webCreateRequiredAction(writer http.ResponseWriter, request *htt
...
 	}
 
 	err = json.Unmarshal(buf, &body)
 	if err != nil {
-		internalErr := &errs.Error{
-			Code: errs.Deserialization,
-		}
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
+		internalErr := errs.NewError(errs.Deserialization, err.Error())
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go:357]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 27]
```diff
@@ -434,7 +352,6 @@ func (a AppAPI) webCreateRequiredAction(writer http.ResponseWriter, request *htt
 
 	internalErr := a.githubRequiredUserActionDao.CreateRequiredUserAction(ct, action)
 	if internalErr != nil {
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/github/app_api.go [hunk 28]
```diff
@@ -417,7 +336,6 @@ func (a AppAPI) webCreateRequiredAction(writer http.ResponseWriter, request *htt
 	genActionIDRes, err := a.cloudClientRegistry.GeneratorClient().GenerateUniqueNumber(ct, genActionIDReq)
 	if err != nil {
 		internalErr := errs.FromGRPCErr(err)
-		a.dataCollector.Logger.ErrorWithContext(ct, internalErr)
 		errs.SetHTTPErr(internalErr, writer)
```
Comment (naturetime10) [inline apps/github/app_api.go]: We do need to log here, or we will lose stack trace.

File: apps/dao/sqldb/github_code_review.go [hunk 29]
```diff
@@ -45,24 +45,14 @@ func (g GithubCodeReview) FindCodeReviewByGithubReviewerID(
...
-			Code: errs.NotFound,
-			Message: fmt.Sprintf(
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return entity.GithubCodeReview{}, errs.NewError(errs.NotFound, fmt.Sprintf(
 				"GithubCodeReview not found: githubReviewerID=%v",
-				githubReviewerID),
+				githubReviewerID))
 		}
-		g.dataCollector.Logger.ErrorWithContext(ct, internalErr)
```
Comment (szheng2207) [inline apps/dao/sqldb/github_code_review.go:55]: I think the log should not be removed?

File: apps/github/app_api.go [hunk 30]
```diff
@@ -1272,15 +1103,7 @@ func (a AppAPI) BackfillPullRequestMetadata(ct context.Context, empty *emptypb.E
...
-		task, rpcErr := a.teamyClientRegistry.TaskClient().GetTask(ct, getTaskReq)
-		if rpcErr != nil {
-			if err == nil {
-				err = errs.FromGRPCErr(rpcErr)
-			}
-
-			a.dataCollector.Logger.ErrorWithContext(ct, errs.FromGRPCErr(rpcErr))
-			continue
-		}
+		task, _ := a.teamyClientRegistry.TaskClient().GetTask(ct, getTaskReq)
```
Comment (naturetime10) [inline apps/github/app_api.go]: Need to handler error

File: apps/github/app_api.go [hunk 31]
```diff
@@ -1293,21 +1116,8 @@ func (a AppAPI) BackfillPullRequestMetadata(ct context.Context, empty *emptypb.E
...
-			a.dataCollector.Logger.ErrorWithContext(ct, gqlErr)
-			continue
-		}
-
-		org, err := a.githubRESTAPI.GetOrganizationByLogin(ct, ins, node.Repository.Owner.Login)
-		if err != nil {
-			a.dataCollector.Logger.ErrorWithContext(ct, err)
-			continue
-		}
+		node, _ := a.githubGraphQLAPI.GetPullRequestByNodeID(ct, ins, pullRequest.NodeID)
```
Comment (naturetime10) [inline apps/github/app_api.go]: Need to handler error

File: apps/github/app_api.go [hunk 32]
```diff
@@ -1293,21 +1116,8 @@ func (a AppAPI) BackfillPullRequestMetadata(ct context.Context, empty *emptypb.E
...
+		org, _ := a.githubRESTAPI.GetOrganizationByLogin(ct, ins, node.Repository.Owner.Login)
```
Comment (naturetime10) [inline apps/github/app_api.go]: Need to handler error

File: apps/github/app_api.go [hunk 33]
```diff
@@ -1319,16 +1129,7 @@ func (a AppAPI) BackfillPullRequestMetadata(ct context.Context, empty *emptypb.E
...
-		if sqlErr != nil {
-			if err == nil {
-				err = sqlErr
-			}
-
-			a.dataCollector.Logger.ErrorWithContext(ct, sqlErr)
-			continue
-		}
-
+		a.githubPullRequestDao.UpdatePullRequest(ct, gpr)
```
Comment (naturetime10) [inline apps/github/app_api.go]: Need to handler error

File: apps/github/app_api.go [hunk 34]
```diff
@@ -1345,23 +1146,17 @@ func (a AppAPI) BackfillPullRequestLinks(ct context.Context, request *appsProto.
...
 			return err
 		}
 
 		for _, prTaskRelation := range prTaskRelations {
-			err = a.tryCreateTaskForPullRequestReviewer(ct, teamID, prEvt.PullRequest.NodeID, prTaskRelation.InternalTaskID, githubReviewer.NodeID, evt, prEvt)
-			if err != nil {
-				a.dataCollector.Logger.ErrorWithContext(ct, err)
-				continue
-			}
+			a.tryCreateTaskForPullRequestReviewer(ct, teamID, prEvt.PullRequest.NodeID, prTaskRelation.InternalTaskID, githubReviewer.NodeID, evt, prEvt)
```
Comment (naturetime10) [inline apps/github/app_api.go]: Need to handler error

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Comment by naturetime10** [comment]
We please format `NewError` also. Thanks.

File: apps/dao/sqldb/github_code_review.go (hunk 29 above)
Comment (duanxianbo) [reply apps/dao/sqldb/github_code_review.go:55]: this one should be removed, we will log it in the api level

File: apps/github/app_api.go (hunk 30 above)
Comment (duanxianbo) [reply apps/github/app_api.go]: you mean to log?

Comment (duanxianbo) [reply apps/github/app_api.go]: log added

**Comment by duanxianbo** [comment]
> NewError

what do you mean by formating newError?

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**